client/post-editor/media-modal/index.jsx

```diff
@@ -111,7 +111,13 @@ export class EditorMediaModal extends Component {
 	isDisabled() {
 		return some( this.props.mediaLibrarySelectedItems, function( item ) {
 			const mimePrefix = MediaUtils.getMimePrefix( item );
-			return item.transient && ( mimePrefix !== 'image' || ModalViews.GALLERY === this.props.view );
+			return item.transient && (
+				// Transients can't be handled by the editor if they are being
+				// uploaded via an external URL
+				item.is_uploading_via_url ||
```

Comment:
I initially thought of merely testing for `typeof item.guid === 'undefined'`, but this felt more explicit: by introducing a new flag, the relationship between these two pieces of the codebase becomes clear with one grep. This would potentially avoid situations where changes to `MediaUtils` suddenly broke `isDisabled`.